landlab/components/uniform_precip/generate_uniform_precip.py

```diff
@@ -26,13 +27,17 @@ class PrecipitationDistribution(Component):
 
     Construction::
 
-        PrecipitationDistribution(mean_storm_duration=0.0,
+        PrecipitationDistribution(grid=None,
+                                  mean_storm_duration=0.0,
                                   mean_interstorm_duration=0.0,
                                   mean_storm_depth=0.0, total_t=0.0,
                                   delta_t=0.0, random_seed=0)
 
     Parameters
     ----------
+    grid : ModelGrid or None
+        A Landlab grid (optional). If provided, storm intensities will be
+        stored as a grid scalar field as the component simulates storms.
     mean_storm_duration : float
```

Comment:
Might also be a possibility to *force* this component to output `at_node`, with the values all the same? Seems like a waste of memory, but 1. the option to not have it do this is still there, and 2. this would be specifically compatible with spatially variable ones.

Let me know what you think? Happy to change this.